{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "cbc7ae3c-3822-40b5-b54c-167aab084061",
   "metadata": {},
   "source": [
    "# Blueberry trio dataQC\n",
    "## HiFi\n",
    "\n",
    "```\n",
    "/input/genomic/plant/Vaccinium/corymbosum/ExperimentRequestor10969_NuixM7_TrioBin/Blueberry_M7xNui/SMRTcell1/CCS_Data\n",
    "/input/genomic/plant/Vaccinium/corymbosum/ExperimentRequestor10969_NuixM7_TrioBin/Blueberry_M7xNui/SMRTcell2/CCS_Data\n",
    "```\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c4190964-5c68-4ba6-b721-399d256e56e2",
   "metadata": {},
   "source": [
    "# Progeny 1 : m64136_221113_041854"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1141d96-6826-4942-a03f-bb052aa36636",
   "metadata": {},
   "source": [
    "## NanoStat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1ba1215b-3ef5-49f9-bebc-8fd4d460dc43",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: NanoStat [-h] [-v] [-o OUTDIR] [-p PREFIX] [-n NAME] [-t N]\n",
      "                [--barcoded] [--readtype {1D,2D,1D2}]\n",
      "                (--fastq file [file ...] | --fasta file [file ...] | --summary file [file ...] | --bam file [file ...] | --ubam file [file ...] | --cram file [file ...])\n",
      "\n",
      "Calculate statistics of long read sequencing dataset.\n",
      "\n",
      "General options:\n",
      "  -h, --help            show the help and exit\n",
      "  -v, --version         Print version and exit.\n",
      "  -o, --outdir OUTDIR   Specify directory for output, only in combination with -n.\n",
      "  -p, --prefix PREFIX   Specify an optional prefix to be used for the output file.\n",
      "  -n, --name NAME       Specify a filename/path for the output, stdout is the default.\n",
      "  -t, --threads N       Set the allowed number of threads to be used by the script.\n",
      "\n",
      "Input options.:\n",
      "  --barcoded            Use if you want to split the summary file by barcode\n",
      "  --readtype {1D,2D,1D2}\n",
      "                        Which read type to extract information about from summary. Options are 1D, 2D,\n",
      "                        1D2\n",
      "\n",
      "Input data sources, one of these is required.:\n",
      "  --fastq file [file ...]\n",
      "                        Data is in one or more (compressed) fastq file(s).\n",
      "  --fasta file [file ...]\n",
      "                        Data is in one or more (compressed) fasta file(s).\n",
      "  --summary file [file ...]\n",
      "                        Data is in one or more (compressed) summary file(s)generated by albacore or\n",
      "                        guppy.\n",
      "  --bam file [file ...]\n",
      "                        Data is in one or more sorted bam file(s).\n",
      "  --ubam file [file ...]\n",
      "                        Data is in one or more unmapped bam file(s).\n",
      "  --cram file [file ...]\n",
      "                        Data is in one or more sorted cram file(s).\n",
      "\n",
      "EXAMPLES:\n",
      "  NanoStat --fastq reads.fastq.gz --outdir statreports\n",
      "  NanoStat --summary sequencing_summary1.txt sequencing_summary2.txtsequencing_summary3.txt --readtype 1D2\n",
      "  NanoStat --bam alignment.bam alignment2.bam\n"
     ]
    }
   ],
   "source": [
    "ml nanopack\n",
    "NanoStat --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f13179a2-b793-436b-b364-aae15f8eeccc",
   "metadata": {},
   "outputs": [],
   "source": [
    "module unload nanopack"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "3ace60af-f190-49cf-be65-6d91af0bcd81",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 1006479\n"
     ]
    }
   ],
   "source": [
    "ml nanopack\n",
    "\n",
    "# set variables for directories\n",
    "BASE=/powerplant/workspace/hrasrb/Blueberry_trio\n",
    "LOG=${BASE}/log\n",
    "WKDIR=${BASE}/data_qc/hifi/m64136_221113_041854\n",
    "OUTPUT=${WKDIR}/02.nanopack\n",
    "\n",
    "# set input file\n",
    "FILE=/powerplant/input/genomic/plant/Vaccinium/corymbosum/ExperimentRequestor10969_NuixM7_TrioBin/Blueberry_M7xNui/SMRTcell1/CCS_Data/*hifi_reads.fastq.gz\n",
    "PREFIX=$(basename ${FILE} .hifi_reads.fastq.gz)\n",
    "\n",
    "# mkdir -pv ${WKDIR} ${OUTPUT}\n",
    "\n",
    "cd $WKDIR\n",
    "\n",
    "# create and submit bash script\n",
    "sbatch << EOF\n",
    "#!/bin/bash -e\n",
    "\n",
    "#SBATCH -J P1_NanoStat\n",
    "#SBATCH --output=${LOG}/hrasrb_%j.out\n",
    "#SBATCH --error=${LOG}/hrasrb_%j.err\n",
    "#SBATCH --mail-user=Sarah.Bailey@plantandfood.co.nz\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH --time=02:00:00 # Walltime\n",
    "#SBATCH --mem=300M\n",
    "#SBATCH --cpus-per-task=2\n",
    "\n",
    "NanoStat --version\n",
    "\n",
    "NanoStat --fastq ${FILE} -t 2 -o . -n ${PREFIX}_NanoStat.out\n",
    "\n",
    "EOF\n",
    "module unload nanopack"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1b8649a6-e3e6-4017-9da5-ee28c10bbed2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# job 1006478\n",
    "# job 1006479"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "73ee33f7-2875-4eac-8c85-2c2833f5e6fe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n",
      "           1006479      fast P1_NanoS   hrasrb  R       0:31      1 aklppb40\n"
     ]
    }
   ],
   "source": [
    "squeue -u hrasrb\n",
    "# seff 1006479"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5f2bfce0-645b-43cd-9308-dac61fcd3a53",
   "metadata": {},
   "source": [
    "## NanoPlot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "9c2f074c-7796-42be-bd8a-75a5c921dc3f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/powerplant/workspace/hrasrb/Blueberry_trio\n",
      "/powerplant/workspace/hrasrb/Blueberry_trio/log\n",
      "/powerplant/workspace/hrasrb/Blueberry_trio/data_qc/hifi/m64136_221113_041854\n",
      "/powerplant/workspace/hrasrb/Blueberry_trio/data_qc/hifi/m64136_221113_041854/02.nanopack\n",
      "/powerplant/input/genomic/plant/Vaccinium/corymbosum/ExperimentRequestor10969_NuixM7_TrioBin/Blueberry_M7xNui/SMRTcell1/CCS_Data/m64136_221113_041854.hifi_reads.fastq.gz\n",
      "m64136_221113_041854\n"
     ]
    }
   ],
   "source": [
    "echo ${BASE}\n",
    "echo ${LOG}\n",
    "echo ${WKDIR}\n",
    "echo ${OUTPUT}\n",
    "echo ${FILE}\n",
    "echo ${PREFIX}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "5880967b-17a2-4020-a5a3-1845f7e432ef",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 1006480\n"
     ]
    }
   ],
   "source": [
    "ml nanopack\n",
    "\n",
    "# create and submit bash script\n",
    "sbatch << EOF\n",
    "#!/bin/bash -e\n",
    "\n",
    "#SBATCH -J P1_NanoPlot\n",
    "#SBATCH --output=${LOG}/hrasrb_%j.out\n",
    "#SBATCH --error=${LOG}/hrasrb_%j.err\n",
    "#SBATCH --mail-user=Sarah.Bailey@plantandfood.co.nz\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH --time=02:00:00 # Walltime\n",
    "#SBATCH --mem=400M\n",
    "#SBATCH --cpus-per-task=2\n",
    "\n",
    "NanoPlot --version\n",
    "\n",
    "echo \"Running on file $PREFIX\"\n",
    "NanoPlot -t 2 --fastq ${FILE} -o ${PREFIX}\n",
    "echo \"Done.\"\n",
    "\n",
    "EOF\n",
    "module unload nanopack"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5a98839c-af2f-48b3-821c-eef1fe124ac5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# job 1006480"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e6058578-76a8-4209-b649-eacc314e0a44",
   "metadata": {},
   "source": [
    "## NanoQC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "635c9633-8b80-4806-8cb4-f3e258aaadad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/powerplant/workspace/hrasrb/Blueberry_trio/data_qc/hifi/m64136_221113_041854\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "dce03b73-496c-4682-bf2f-12900cf19fa3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) Submitted batch job 1006485\n",
      "(/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) "
     ]
    }
   ],
   "source": [
    "ml conda/4.10.3\n",
    "conda activate hrasrb_nanoQC\n",
    "\n",
    "# set variables for directories\n",
    "BASE=/powerplant/workspace/hrasrb/Blueberry_trio\n",
    "LOG=${BASE}/log\n",
    "WKDIR=${BASE}/data_qc/hifi/m64136_221113_041854/02.nanopack\n",
    "\n",
    "# set input file\n",
    "FILE=/powerplant/input/genomic/plant/Vaccinium/corymbosum/ExperimentRequestor10969_NuixM7_TrioBin/Blueberry_M7xNui/SMRTcell1/CCS_Data/*hifi_reads.fastq.gz\n",
    "\n",
    "# mkdir -pv ${WKDIR} ${OUTPUT}\n",
    "\n",
    "cd $WKDIR\n",
    "\n",
    "# create and submit bash script\n",
    "sbatch << EOF\n",
    "#!/bin/bash -e\n",
    "\n",
    "#SBATCH -J P1_NanoQC\n",
    "#SBATCH --output=${LOG}/hrasrb_%j.out\n",
    "#SBATCH --error=${LOG}/hrasrb_%j.err\n",
    "#SBATCH --mail-user=Sarah.Bailey@plantandfood.co.nz\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH --time=02:00:00 # Walltime\n",
    "#SBATCH --mem=400M\n",
    "\n",
    "nanoQC -v\n",
    "\n",
    "nanoQC -o . ${FILE}\n",
    "\n",
    "EOF\n",
    "\n",
    "conda deactivate\n",
    "module unload conda"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "e6b5c5d5-60b7-4f5e-93e5-fbe1897f0d19",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job ID: 1006484\n",
      "Cluster: powerplant\n",
      "User/Group: hrasrb/hrasrb\n",
      "State: FAILED (exit code 127)\n",
      "Cores: 1\n",
      "CPU Utilized: 00:00:00\n",
      "CPU Efficiency: 0.00% of 00:00:00 core-walltime\n",
      "Job Wall-clock time: 00:00:00\n",
      "Memory Utilized: 0.00 MB (estimated maximum)\n",
      "Memory Efficiency: 0.00% of 400.00 MB (400.00 MB/node)\n"
     ]
    }
   ],
   "source": [
    "# job 1006481\n",
    "# job 1006484\n",
    "# job 1006485\n",
    "seff 1006485"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "50560ccd-df77-4320-b2f8-171864c6b522",
   "metadata": {},
   "source": [
    "# Progeny 2 : m64136_221114_133241"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1b8c4008-7778-46ed-ba75-5f11159f4c20",
   "metadata": {},
   "source": [
    "## NanoStat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "41d2952f-2078-4463-9e81-aab7fb1854d2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: created directory ‘/powerplant/workspace/hrasrb/Blueberry_trio/data_qc/hifi/m64136_221114_133241/02.nanopack’\n",
      "Submitted batch job 1006482\n"
     ]
    }
   ],
   "source": [
    "ml nanopack\n",
    "\n",
    "# set variables for directories\n",
    "BASE=/powerplant/workspace/hrasrb/Blueberry_trio\n",
    "LOG=${BASE}/log\n",
    "WKDIR=${BASE}/data_qc/hifi/m64136_221114_133241/02.nanopack\n",
    "\n",
    "# set input file\n",
    "FILE=/powerplant/input/genomic/plant/Vaccinium/corymbosum/ExperimentRequestor10969_NuixM7_TrioBin/Blueberry_M7xNui/SMRTcell2/CCS_Data/*hifi_reads.fastq.gz\n",
    "PREFIX=$(basename ${FILE} .hifi_reads.fastq.gz)\n",
    "\n",
    "mkdir -pv ${WKDIR}\n",
    "\n",
    "cd $WKDIR\n",
    "\n",
    "# create and submit bash script\n",
    "sbatch << EOF\n",
    "#!/bin/bash -e\n",
    "\n",
    "#SBATCH -J P2_NanoStat\n",
    "#SBATCH --output=${LOG}/hrasrb_%j.out\n",
    "#SBATCH --error=${LOG}/hrasrb_%j.err\n",
    "#SBATCH --mail-user=Sarah.Bailey@plantandfood.co.nz\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH --time=02:00:00 # Walltime\n",
    "#SBATCH --mem=400M\n",
    "\n",
    "NanoStat --version\n",
    "\n",
    "NanoStat --fastq ${FILE} -t 1 -o . -n ${PREFIX}_NanoStat.out\n",
    "\n",
    "EOF\n",
    "module unload nanopack"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6292f013-5429-4d16-b8ed-ab5bc77aff7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# job 1006482"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "cd5c5da0-b733-43f3-8eb9-a2c95fa2da63",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n",
      "           1006482      fast P2_NanoS   hrasrb  R       0:09      1 aklppb40\n",
      "           1006480      fast P1_NanoP   hrasrb  R       6:01      1 aklppb40\n",
      "           1006479      fast P1_NanoS   hrasrb  R       7:24      1 aklppb40\n"
     ]
    }
   ],
   "source": [
    "squeue -u hrasrb\n",
    "# seff 1006482"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9224d88f-137f-4222-8869-e9ad95abbc4d",
   "metadata": {},
   "source": [
    "## NanoPlot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "c07a24af-ca86-4a4c-9612-f2ac74afa6cc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/powerplant/workspace/hrasrb/Blueberry_trio/data_qc/hifi/m64136_221114_133241/02.nanopack\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "462acd1c-7748-4793-b9d5-354c1df6c340",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/powerplant/workspace/hrasrb/Blueberry_trio\n",
      "/powerplant/workspace/hrasrb/Blueberry_trio/log\n",
      "/powerplant/workspace/hrasrb/Blueberry_trio/data_qc/hifi/m64136_221114_133241/02.nanopack\n",
      "/powerplant/input/genomic/plant/Vaccinium/corymbosum/ExperimentRequestor10969_NuixM7_TrioBin/Blueberry_M7xNui/SMRTcell2/CCS_Data/m64136_221114_133241.hifi_reads.fastq.gz\n",
      "m64136_221114_133241\n"
     ]
    }
   ],
   "source": [
    "echo ${BASE}\n",
    "echo ${LOG}\n",
    "echo ${WKDIR}\n",
    "echo ${FILE}\n",
    "echo ${PREFIX}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "a45e3c2b-f926-49f4-8ff5-986db3e3e1cf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 1006483\n"
     ]
    }
   ],
   "source": [
    "ml nanopack\n",
    "\n",
    "# create and submit bash script\n",
    "sbatch << EOF\n",
    "#!/bin/bash -e\n",
    "\n",
    "#SBATCH -J P2_NanoPlot\n",
    "#SBATCH --output=${LOG}/hrasrb_%j.out\n",
    "#SBATCH --error=${LOG}/hrasrb_%j.err\n",
    "#SBATCH --mail-user=Sarah.Bailey@plantandfood.co.nz\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH --time=02:00:00 # Walltime\n",
    "#SBATCH --mem=400M\n",
    "#SBATCH --cpus-per-task=2\n",
    "\n",
    "NanoPlot --version\n",
    "\n",
    "echo \"Running on file $PREFIX\"\n",
    "NanoPlot -t 2 --fastq ${FILE} -o ${PREFIX}\n",
    "echo \"Done.\"\n",
    "\n",
    "EOF\n",
    "module unload nanopack"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "44aaf7e8-c5f8-4140-a37a-e0f227bd9aa1",
   "metadata": {},
   "outputs": [],
   "source": [
    "# job 1006483"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "26740527-68fa-4411-932d-4ac91eae1ce0",
   "metadata": {},
   "source": [
    "## NanoQC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e8d6d70a-482b-4f15-8f87-8cea38520851",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/powerplant/workspace/hrasrb/Blueberry_trio/data_qc/hifi/m64136_221114_133241/02.nanopack\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "fee29bb5-5870-418c-82d6-5ecedfc95f9b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) Submitted batch job 1006486\n",
      "(/workspace/appscratch/miniconda/hrasrb_nanoQC) (/workspace/appscratch/miniconda/hrasrb_nanoQC) "
     ]
    }
   ],
   "source": [
    "ml conda/4.10.3\n",
    "conda activate hrasrb_nanoQC\n",
    "\n",
    "# set variables for directories\n",
    "BASE=/powerplant/workspace/hrasrb/Blueberry_trio\n",
    "LOG=${BASE}/log\n",
    "WKDIR=${BASE}/data_qc/hifi/m64136_221114_133241/02.nanopack\n",
    "\n",
    "# set input file\n",
    "FILE=/powerplant/input/genomic/plant/Vaccinium/corymbosum/ExperimentRequestor10969_NuixM7_TrioBin/Blueberry_M7xNui/SMRTcell2/CCS_Data/*hifi_reads.fastq.gz\n",
    "\n",
    "# mkdir -pv ${WKDIR} ${OUTPUT}\n",
    "\n",
    "cd $WKDIR\n",
    "\n",
    "# create and submit bash script\n",
    "sbatch << EOF\n",
    "#!/bin/bash -e\n",
    "\n",
    "#SBATCH -J P2_NanoQC\n",
    "#SBATCH --output=${LOG}/hrasrb_%j.out\n",
    "#SBATCH --error=${LOG}/hrasrb_%j.err\n",
    "#SBATCH --mail-user=Sarah.Bailey@plantandfood.co.nz\n",
    "#SBATCH --mail-type=ALL\n",
    "#SBATCH --time=02:00:00 # Walltime\n",
    "#SBATCH --mem=400M\n",
    "\n",
    "nanoQC -v\n",
    "\n",
    "nanoQC -o . ${FILE}\n",
    "\n",
    "EOF\n",
    "\n",
    "conda deactivate\n",
    "module unload conda"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d3b7a6f7-be37-4d58-ac91-d2920960529a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# job 1006486\n",
    "seff 1006486"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "396a5e9c-8d87-4df5-824c-fe04bc2cc8dc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)\n",
      "           1006486      fast P2_NanoQ   hrasrb  R       0:19      1 aklppb40\n",
      "           1006485      fast P1_NanoQ   hrasrb  R       2:03      1 aklppb40\n",
      "           1006483      fast P2_NanoP   hrasrb  R      13:19      1 aklppb40\n",
      "           1006482      fast P2_NanoS   hrasrb  R      15:00      1 aklppb40\n",
      "           1006480      fast P1_NanoP   hrasrb  R      20:52      1 aklppb40\n",
      "           1006479      fast P1_NanoS   hrasrb  R      22:15      1 aklppb40\n"
     ]
    }
   ],
   "source": [
    "squeue -u hrasrb"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
