{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "ed696821-0dc1-4160-895c-186038dd2075",
   "metadata": {},
   "source": [
    "# Hi-C Scaffolding Urchin\n",
    "### New assembly needs new scaffolding"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "918d0e0c-0155-471c-a4fc-31201ea1fcd3",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Set Up\n",
    "WKDIR=/workspace/hraijc/HiC_trials/HiC23/NovaSeq/Urchin/Crodergsii_V3\n",
    "TEMP=${WKDIR}/TEMP\n",
    "ASSEMBLY=/workspace/hraijc/HiC_trials/HiC23/Assemblies/urchin_230923_primary_purged.fa\n",
    "#ASSEMBLY_POSTSCAFF=/workspace/hraijc/HiC_trials/HiC23/Results/Crodergsii_HiC092623_v3.fa\n",
    "HiC_R1=/workspace/hraijc/HiC_trials/HiC23/NovaSeq/Urchin/Crodergsii_V3/reads/Urchin_HiC_combined_filtered_R1.fq.gz\n",
    "HiC_R2=/workspace/hraijc/HiC_trials/HiC23/NovaSeq/Urchin/Crodergsii_V3/reads/Urchin_HiC_combined_filtered_R2.fq.gz\n",
    "OUT_PREFIX=Crodergsii_HiC092623\n",
    "#TELOMERE=TTAGGG #http://telomerase.asu.edu/sequences_telomere.html\n",
    "#GENOME_SIZE=880000000\n",
    "#NF_CONFIG=/workspace/hraijc/HiC_trials/HiC23/NovaSeq/Urchin/hic-scaffolding-pipeline/nextflow.config\n",
    "\n",
    "mkdir -p $WKDIR/log\n",
    "mkdir -p $TEMP\n",
    "cd $WKDIR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7c2f2972-2160-47d2-8f9d-83d4338b0014",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c5cb93d7-ddd6-4e78-b600-a085d0437592",
   "metadata": {},
   "outputs": [],
   "source": [
    "#cp /input/globus-test/transfer/public/hraijc/urchin_092023/*.fa /workspace/hraijc/HiC_trials/HiC23/Assemblies/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f3073bdb-615a-48ad-8ec3-5ba7030c70ae",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Using Annabels filtered reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d0d9237d-10b4-43c5-880a-20811d7a9d2c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Prep Reference\n",
    "\n",
    "sbatch << EOF\n",
    "#!/bin/bash\n",
    "#SBATCH -J bwaindex\n",
    "#SBATCH --cpus-per-task=1\n",
    "#SBATCH --mem=1G\n",
    "#SBATCH --time=1:00:00\n",
    "#SBATCH -o ${WKDIR}/log/%J.out\n",
    "#SBATCH -e ${WKDIR}/log/%J.err\n",
    "\n",
    "module load bwa/0.7.17\n",
    "module load samtools/1.16\n",
    "\n",
    "\n",
    "samtools faidx ${ASSEMBLY}\n",
    "bwa index ${ASSEMBLY}\n",
    "\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "21c984b2-3d77-44c6-b5f0-7579be0831f6",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Map HiC reads\n",
    "\n",
    "sbatch << EOF\n",
    "#!/bin/bash\n",
    "#SBATCH -J hic\n",
    "#SBATCH -o ${WKDIR}/log/%J.out\n",
    "#SBATCH -e ${WKDIR}/log/%J.err\n",
    "#SBATCH --cpus-per-task=12\n",
    "#SBATCH --mem=20G\n",
    "#SBATCH --time=73:00:00\n",
    "\n",
    "module load bwa/0.7.17\n",
    "module load samtools/1.16\n",
    "\n",
    "\n",
    "cd ${WKDIR}\n",
    "\n",
    "### Align reads with BWA. Use -5SP for Hi-C reads.#############################\n",
    "\n",
    "bwa mem -5SP -t12 ${ASSEMBLY} ${HiC_R1} ${HiC_R2} -o ${TEMP}/${OUT_PREFIX}.sam\n",
    "\n",
    "\n",
    "### Flag PCR Duplicates with SAMBLASTER #######################################\n",
    "/workspace/hraijc/git_clones/samblaster/samblaster -i ${TEMP}/${OUT_PREFIX}.sam -o ${TEMP}/${OUT_PREFIX}_marked_byread.sam\n",
    "\n",
    "\n",
    "### Remove unmmaped and non-primary aligned reads. Sort and Index bam files.###\n",
    "samtools view -S -b -h -@ 12 -F 2316 ${TEMP}/${OUT_PREFIX}_marked_byread.sam > ${TEMP}/${OUT_PREFIX}_presort_marked.bam\n",
    "samtools sort -@ 12 ${TEMP}/${OUT_PREFIX}_presort_marked.bam -o ${OUT_PREFIX}_marked.bam\n",
    "samtools view -S -b -h -@ 12 -F 3340 ${TEMP}/${OUT_PREFIX}_marked_byread.sam > ${TEMP}/${OUT_PREFIX}_presort_dedup.bam\n",
    "samtools sort -@ 12 ${TEMP}/${OUT_PREFIX}_presort_dedup.bam -o ${OUT_PREFIX}_dedup.bam\n",
    "\n",
    "\n",
    "### Run YAHS with contig and scaffolding error correction. ####################\n",
    "/workspace/hraijc/git_clones/yahs/yahs --no-mem-check ${ASSEMBLY} ${OUT_PREFIX}_dedup.bam -o ${OUT_PREFIX}_dedup_yahs3\n",
    "\n",
    "EOF\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "226b105d-05e3-42d4-9a84-6487bf7c1f46",
   "metadata": {},
   "outputs": [],
   "source": [
    "out=${OUT_PREFIX}_dedup_yahs3 # prefix of outfiles produced by YAHS.  \n",
    "contigs=${ASSEMBLY} # need to be indexed, i.e., test.fa.gz.fai in same directory\n",
    "wkdir=${WKDIR}\n",
    "\n",
    "\n",
    "sbatch --job-name=hic_mapyahs \\\n",
    "    -o ${WKDIR}/log/%J.out \\\n",
    "    -e ${WKDIR}/log/%J.err \\\n",
    "    --cpus-per-task=8 \\\n",
    "    --mem=24G \\\n",
    "    --time=03:10:00 \\\n",
    "    --export=out=$out,contigs=$contigs,wkdir=$wkdir \\\n",
    "    /workspace/hraijc/Gitrepos/High-quality-genomes/Methods/DNase_HiC/notebooks/yahs_contactmapgen2.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4db7a8b0-b13f-40d5-8ab1-56f92ea1a9dc",
   "metadata": {},
   "outputs": [],
   "source": [
    "seff 2069748\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8cf08b62-0555-4391-bd3a-4a563cc18282",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Re-Run YAHS at higher resolutions to fix breaks in smaller contigs. \n",
    "\n",
    "sbatch << EOF\n",
    "#!/bin/bash\n",
    "#SBATCH -J hic\n",
    "#SBATCH -o ${WKDIR}/log/%J.out\n",
    "#SBATCH -e ${WKDIR}/log/%J.err\n",
    "#SBATCH --cpus-per-task=1\n",
    "#SBATCH --mem=20G\n",
    "#SBATCH --time=6:00:00\n",
    "\n",
    "\n",
    "\n",
    "cd ${WKDIR}\n",
    "\n",
    "### Run YAHS with contig and scaffolding error correction. ####################\n",
    "#/workspace/hraijc/git_clones/yahs/yahs --no-mem-check ${ASSEMBLY} ${OUT_PREFIX}_dedup.bam -o ${OUT_PREFIX}_dedup_yahs3\n",
    "/workspace/hraijc/git_clones/yahs/yahs --no-mem-check ${ASSEMBLY} ${OUT_PREFIX}_dedup.bam -r 5000,10000,20000,50000,100000,200000,500000,1000000,2000000,5000000,10000000,20000000,50000000,100000000,200000000,500000000 -o ${OUT_PREFIX}_dedup_yahs3_5k\n",
    "#/workspace/hraijc/git_clones/yahs/yahs --no-mem-check ${ASSEMBLY} ${OUT_PREFIX}_dedup.bam -r 1000,5000,10000,20000,50000,100000,200000,500000,1000000,2000000,5000000,10000000,20000000,50000000,100000000,200000000,500000000 -o ${OUT_PREFIX}_dedup_yahs3_1k\n",
    "\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "24132051-c0f6-4cde-bba3-07bb29ccaa42",
   "metadata": {},
   "outputs": [],
   "source": [
    "out=${OUT_PREFIX}_dedup_yahs3_5k # prefix of outfiles produced by YAHS.  \n",
    "contigs=${ASSEMBLY} # need to be indexed, i.e., test.fa.gz.fai in same directory\n",
    "wkdir=${WKDIR}\n",
    "\n",
    "\n",
    "sbatch --job-name=hic_mapyahs \\\n",
    "    -o ${WKDIR}/log/%J.out \\\n",
    "    -e ${WKDIR}/log/%J.err \\\n",
    "    --cpus-per-task=8 \\\n",
    "    --mem=24G \\\n",
    "    --time=03:10:00 \\\n",
    "    --export=out=$out,contigs=$contigs,wkdir=$wkdir \\\n",
    "    /workspace/hraijc/Gitrepos/High-quality-genomes/Methods/DNase_HiC/notebooks/yahs_contactmapgen2.sh\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dcadeed6-57ef-4331-a228-d06a7b0dd09e",
   "metadata": {},
   "outputs": [],
   "source": [
    "##############\n",
    "#Do manual scaffolding cleanup in Juicer then run\n",
    "\n",
    "##############\n",
    "\n",
    "\n",
    "sbatch << EOF\n",
    "#!/bin/bash\n",
    "#SBATCH -J YAHS_liftover\n",
    "#SBATCH -o ${WKDIR}/log/%J.out\n",
    "#SBATCH -e ${WKDIR}/log/%J.err\n",
    "#SBATCH --cpus-per-task=1\n",
    "#SBATCH --mem=4G\n",
    "#SBATCH --time=2:00:00\n",
    "\n",
    "\n",
    "\n",
    "cd ${WKDIR}\n",
    "\n",
    "\n",
    "\n",
    "/workspace/hraijc/git_clones/yahs/juicer post -o Crodergsii_HiC092623_dedup_yahs3_5k_man1 Crodergsii_HiC092623_dedup_yahs3_5k_JBAT.review.assembly Crodergsii_HiC092623_dedup_yahs3_5k_JBAT.liftover.agp ${ASSEMBLY}\n",
    "\n",
    "\n",
    "\n",
    "EOF\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1505f981-e28f-4584-a9e5-0a594b9b2e97",
   "metadata": {},
   "outputs": [],
   "source": [
    "# SET INFO FOR SUBSETTING UNIQUE MAPPED HIC READS\n",
    "WKDIR=/workspace/hraijc/HiC_trials/HiC23/NovaSeq/Urchin/Crodergsii_V3\n",
    "READ1=${WKDIR}/reads/Urchin_HiC_U_R1.fq.gz\n",
    "READ2=${WKDIR}/reads/Urchin_HiC_U_R2.fq.gz\n",
    "DEDUP_BAM=${WKDIR}/Crodergsii_HiC092623_dedup.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71a87ed3-e697-47f5-a3c7-03e3e307f5d0",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "#Subset HiC reads\n",
    "\n",
    "sbatch << EOF\n",
    "#!/bin/bash\n",
    "#SBATCH -J hic_subset\n",
    "#SBATCH -o ${WKDIR}/log/%J.out\n",
    "#SBATCH -e ${WKDIR}/log/%J.err\n",
    "#SBATCH --cpus-per-task=12\n",
    "#SBATCH --mem=12G\n",
    "#SBATCH --time=2:00:00\n",
    "\n",
    "\n",
    "module load samtools/1.16\n",
    "\n",
    "\n",
    "cd ${WKDIR}\n",
    "samtools sort -@ 12 -n ${DEDUP_BAM} -o ${DEDUP_BAM%.bam}_namesort.bam\n",
    "\n",
    "samtools bam2fq -@ 12 -1 ${READ1} -2 ${READ2} ${DEDUP_BAM%.bam}_namesort.bam\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7eee43c0-4341-4cff-bc25-c114b3e342b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "sbatch << EOF\n",
    "#!/bin/bash\n",
    "#SBATCH -J minimap\n",
    "#SBATCH -o ${WKDIR}/log/%J.out\n",
    "#SBATCH -e ${WKDIR}/log/%J.err\n",
    "#SBATCH --cpus-per-task=4\n",
    "#SBATCH --mem=2G\n",
    "#SBATCH --time=01:10:00\n",
    "\n",
    "cd /workspace/hraijc/HiC_trials/HiC23/NovaSeq/Urchin/Crodergsii_V3/Synteny\n",
    "\n",
    "module load minimap2\n",
    "\n",
    "minimap2 -t 4 -cx asm10 Crodergsii_HiC092623_dedup_yahs3_5k_scaffolds_final.fa Crodergsii_HiC092623_dedup_yahs3_5k_man1.fa > UrchinYahs5_to_man1.paf \n",
    "EOF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f3360725-7e28-4796-887a-ff1dbf1bc0c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Run YAHS without scaffolding just to generate agp assembly and Hic files. \n",
    "\n",
    "sbatch << EOF\n",
    "#!/bin/bash\n",
    "#SBATCH -J hic\n",
    "#SBATCH -o ${WKDIR}/log/%J.out\n",
    "#SBATCH -e ${WKDIR}/log/%J.err\n",
    "#SBATCH --cpus-per-task=1\n",
    "#SBATCH --mem=20G\n",
    "#SBATCH --time=6:00:00\n",
    "\n",
    "\n",
    "\n",
    "cd ${WKDIR}\n",
    "\n",
    "### Run YAHS with contig and scaffolding error correction. ####################\n",
    "#/workspace/hraijc/git_clones/yahs/yahs --no-mem-check ${ASSEMBLY} ${OUT_PREFIX}_dedup.bam -o ${OUT_PREFIX}_dedup_yahs3\n",
    "/workspace/hraijc/git_clones/yahs/yahs --no-mem-check ${ASSEMBLY} ${OUT_PREFIX}_dedup.bam -r 5000,10000,20000,50000,100000,200000,500000,1000000,2000000,5000000,10000000,20000000,50000000,100000000,200000000,500000000 -o ${OUT_PREFIX}_dedup_yahs3_5k\n",
    "/workspace/hraijc/git_clones/yahs/yahs --no-mem-check ${ASSEMBLY} ${OUT_PREFIX}_dedup.bam -r 1000,5000,10000,20000,50000,100000,200000,500000,1000000,2000000,5000000,10000000,20000000,50000000,100000000,200000000,500000000 -o ${OUT_PREFIX}_dedup_yahs3_1k\n",
    "\n",
    "\n",
    "EOF"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
